drivers/adc/adc_smartbond_sdadc.c

```diff
@@ -107,6 +115,45 @@ static int sdadc_smartbond_channel_setup(const struct device *dev,
 				     SDADC_SDADC_CTRL_REG_SDADC_SE_Msk		\
 )
 
+static inline void sdadc_smartbond_pm_policy_state_lock_get(const struct device *dev,
+					      struct sdadc_smartbond_data *data)
+{
+#if defined(CONFIG_PM_DEVICE_RUNTIME)
+	pm_device_runtime_get(dev);
+
+#endif
+
+#if defined(CONFIG_PM_DEVICE)
```

Comment:
This safeguard should be moved at the beginning (as done [here](https://github.com/zephyrproject-rtos/zephyr/pull/73505/files#diff-bf44e7d7d42cd51eb2e0c743e7f9932258bcb0559119d191991dfa1835cb5629R135)). The same goes for the `*_put` subroutine.

Reply:
Done